js/pihole/index.js

```diff
@@ -223,8 +223,9 @@ function updateQueryTypes() {
 function updateTopClientsChart() {
     $.getJSON("api.php?summaryRaw&getQuerySources", function(data) {
         var clienttable =  $('#client-frequency').find('tbody:last');
-        for (domain in data.top_sources) {
-            clienttable.append('<tr> <td>' + domain +
+        for (var domain in data.top_sources) {
+            var url = url = "<a href=\"queries.php?client="+domain+"\">"+domain+"</a>";
```

Comment:
Sanitize the domain, since it can be subject to an XSS attack. The domain could be valid JS/HTML code, which would cause the client's browser to execute it. Maybe something like [this](http://stackoverflow.com/a/4835406)?